cocotb/share/lib/simulator/simulatormodule.cpp

```diff
@@ -921,138 +921,164 @@ PyMODINIT_FUNC PyInit_simulator(void)
     return simulator;
 }
 
+/* NOTE: in the following docstrings we are specifying the parameters twice, but this is necessary.
+ * The first docstring before the long '--' line specifies the __text_signature__ that is used
+ * by the help() function. And the second after the '--' line contains type annotations used by
+ * the `autodoc_docstring_signature` setting of sphinx.ext.autodoc for generating documentation
+ * because type annotations are not supported in __text_signature__.
+ */
+
 static PyMethodDef gpi_sim_hdl_methods[] = {
     {"get_signal_val_long",
         (PyCFunction)get_signal_val_long, METH_NOARGS, PyDoc_STR(
             "get_signal_val_long($self)\n"
             "--\n\n"
-            "Get the value of a signal as a long"
+            "get_signal_val_long(self) -> int\n"
+            "Get the value of a signal as an integer."
         )
     },
     {"get_signal_val_str",
         (PyCFunction)get_signal_val_str, METH_NOARGS, PyDoc_STR(
             "get_signal_val_str($self)\n"
             "--\n\n"
-            "Get the value of a signal as an ASCII string"
+            "get_signal_val_str(self) -> bytes\n"
+            "Get the value of a signal as a byte string."
         )
     },
     {"get_signal_val_binstr",
         (PyCFunction)get_signal_val_binstr, METH_NOARGS, PyDoc_STR(
             "get_signal_val_binstr($self)\n"
             "--\n\n"
-            "Get the value of a signal as a binary string"
+            "get_signal_val_binstr(self) -> str\n"
+            "Get the value of a logic vector signal as a string of (``0``, ``1``, ``X``, etc.), one element per character."
         )
     },
     {"get_signal_val_real",
         (PyCFunction)get_signal_val_real, METH_NOARGS, PyDoc_STR(
             "get_signal_val_real($self)\n"
             "--\n\n"
-            "Get the value of a signal as a double precision float"
+            "get_signal_val_real(self) -> float\n"
+            "Get the value of a signal as a float."
         )
     },
     {"set_signal_val_long",
         (PyCFunction)set_signal_val_long, METH_VARARGS, PyDoc_STR(
             "set_signal_val_long($self, action, value, /)\n"
             "--\n\n"
-            "Set the value of a signal using a long"
+            "set_signal_val_long(self, action: int, value: int) -> None\n"
+            "Set the value of a signal using an integer.\n"
         )
     },
     {"set_signal_val_str",
         (PyCFunction)set_signal_val_str, METH_VARARGS, PyDoc_STR(
             "set_signal_val_str($self, action, value, /)\n"
             "--\n\n"
-            "Set the value of a signal using an NUL-terminated 8-bit string"
+            "set_signal_val_str(self, action: int, value: bytes) -> None\n"
```

Comment:
Just realized the docs don't render correctly - the `self` argument should not appear. I think this should be either
```suggestion
            "set_signal_val_str(action: int, value: bytes) -> None\n"
```
or

```suggestion
            "hdl.set_signal_val_str(action: int, value: bytes) -> None\n"
```

Mind doing a follow-up PR?

Reply:
Sure.